src/webots/gui/WbConsole.cpp

```diff
@@ -410,54 +418,89 @@ void WbConsole::handlePossibleAnsiEscapeSequences(const QString &msg, WbLog::Lev
           mBold = false;
         } else if (code == "1")  // bold
           mBold = true;
-        else if (code.startsWith("3")) {  // foreground color change
+        else if (code.startsWith("2")) {
+          const char c = code.toLocal8Bit().data()[1];
+          if (c == 'J') {
+            WbLog::clear();
+            html.clear();  // nothing to output since clear has been done
+          }
+        } else if (code.startsWith("3")) {  // foreground color change
+          const char c = code.toLocal8Bit().data()[1];
+          switch (c) {
+            case '1':
+              mFgColor = ansiRed();
+              break;
+            case '2':
+              mFgColor = ansiGreen();
+              break;
+            case '3':
+              mFgColor = ansiYellow();
+              break;
+            case '4':
+              mFgColor = ansiBlue();
+              break;
+            case '5':
+              mFgColor = ansiMagenta();
+              break;
+            case '6':
+              mFgColor = ansiCyan();
+              break;
+            case '7':
+              mFgColor = ansiWhite();
+              break;
+            case '9':  // 39 - Default text color
+            default:
+              mFgColor = ansiBlack();
+              break;
+          }
+        } else if (code.startsWith("4")) {  // background color change
           const char c = code.toLocal8Bit().data()[1];
           switch (c) {
             case '1':
-              mColor = ansiRed();
+              mBgColor = ansiRed();
               break;
             case '2':
-              mColor = ansiGreen();
+              mBgColor = ansiGreen();
               break;
             case '3':
-              mColor = ansiYellow();
+              mBgColor = ansiYellow();
               break;
             case '4':
-              mColor = ansiBlue();
+              mBgColor = ansiBlue();
               break;
             case '5':
-              mColor = ansiMagenta();
+              mBgColor = ansiMagenta();
               break;
             case '6':
-              mColor = ansiCyan();
+              mBgColor = ansiCyan();
               break;
             case '7':
-              mColor = ansiWhite();
+              mBgColor = ansiWhite();
               break;
             case '9':  // 39 - Default text color
             default:
-              mColor = ansiBlack();
+              mBgColor = ansiBlack();
               break;
           }
         }
       }
-
       int j = i;
       i = msg.indexOf("\033[", i);
-      if (i == -1) {
+      if (i == -1) {  // Previous escape code was the last one found
         const QString remains(msg.mid(j));
         html += htmlSpan(remains, level);
         handleCRAndLF(html);
         return;
       }
-      if (j != i)
+      if (j != i) {  // Extract text contained between two escape codes if so
         html += htmlSpan(msg.mid(j, i - j), level);
+      }
     }
-    handleCRAndLF(html);
```

Comment:
This was never used since we return in the above if when we found the last escape character, otherwise if there was not in the raw string, we pass in the underneath else.

Reply:
Well spotted. It seems we could even remove the following `else`.